app/src/server/tasks/evaluateTestSetEntries.task.ts

```diff
@@ -140,14 +165,15 @@ export const evaluateTestSetEntries = defineTask<EvalKey>({
         join
           .onRef("der.nodeEntryInputHash", "=", "ne.inputHash")
           .on("der.datasetEvalOutputSourceId", "=", firstOutputSourceId)
-          .on("der.status", "=", "COMPLETE"),
+          .on("der.status", "=", "COMPLETE")
+          .on("der.score", "!=", null),
       )
       .innerJoin("DatasetEvalResult as comparisonDer", (join) =>
         join
           .onRef("comparisonDer.nodeEntryInputHash", "=", "ne.inputHash")
-          .on("comparisonDer.nodeEntryOutputHash", "=", nodeEntry.outputHash)
           .on("comparisonDer.datasetEvalOutputSourceId", "=", secondOutputSourceId)
-          .on("der.status", "=", "COMPLETE"),
```

Comment:
This line was the primary cause of the bug. Instead of checking if `comparisonDer` was complete, we were only checking to see if `der` was complete, which might have been a DatasetEvalResult that was not compared against`comparisonDer`. This meant that in many cases, `comparisonDer` was not actually complete, so the score we copied from it was null.